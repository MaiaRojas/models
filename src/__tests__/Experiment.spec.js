const mongoose = require('mongoose');
const { Experiment } = require('../../')(mongoose);

describe('Experiment', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
  });

  it('should fail with missing props', () => {
    const experiment = new Experiment();
    return experiment.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail when experiment has fields missing', () => {
    const experiment = new Experiment({
      uid: '45orQ7aPbbPvF2fvMgbA1TsTFPo2',
      createdBy: mongoose.Types.ObjectId('5ca30d9cdd9c6f29c7d7b591'),
      slug: 'text',
      name: 'text',
    });
    return experiment.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const experiment = new Experiment({
      uid: '45orQ7aPbbPvF2fvMgbA1TsTFPo2',
      createdBy: mongoose.Types.ObjectId('5ca30d9cdd9c6f29c7d7b591'),
      slug: 'text',
      name: 'text',
      description: 'prueba text',
    });

    return experiment.save()
      .then((result) => {
        expect(typeof result).toBe('object');
        expect(result.name).toBe(experiment.name);
      });
  });
});
