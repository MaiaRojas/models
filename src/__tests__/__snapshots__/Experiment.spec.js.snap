// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiment should fail when experiment has fields missing 1`] = `
Object {
  "description": [ValidatorError: Path \`description\` is required.],
}
`;

exports[`Experiment should fail with missing props 1`] = `
Object {
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "description": [ValidatorError: Path \`description\` is required.],
  "name": [ValidatorError: Path \`name\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
  "uid": [ValidatorError: Path \`uid\` is required.],
}
`;
